commitrmnocb/outcome.go

```diff
@@ -0,0 +1,316 @@
+package commitrmnocb
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+	"golang.org/x/exp/maps"
+
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+)
+
+// Outcome depending on the current state, either:
+// - chooses the seq num ranges for the next round
+// - builds a report
+// - checks for the transmission of a previous report
+func (p *Plugin) Outcome(
+	outCtx ocr3types.OutcomeContext, query types.Query, aos []types.AttributedObservation,
+) (ocr3types.Outcome, error) {
+	previousOutcome, nextState := p.decodeOutcome(outCtx.PreviousOutcome)
+	commitQuery := Query{}
+
+	consensusObservation, err := p.getConsensusObservation(aos)
+	if err != nil {
+		return ocr3types.Outcome{}, err
+	}
+
+	outcome := Outcome{}
+
+	switch nextState {
+	case SelectingRangesForReport:
+		outcome = p.ReportRangesOutcome(commitQuery, consensusObservation)
+
+	case BuildingReport:
+		outcome = p.buildReport(commitQuery, consensusObservation)
+
+	case WaitingForReportTransmission:
+		outcome = p.checkForReportTransmission(previousOutcome, consensusObservation)
+
+	default:
+		p.lggr.Warnw("Unexpected state in Outcome", "state", nextState)
+		return outcome.Encode()
+	}
+
+	p.lggr.Infow("Commit Plugin Outcome", "outcome", outcome, "oid", p.nodeID)
+	return outcome.Encode()
+}
+
+// ReportRangesOutcome determines the sequence number ranges for each chain to build a report from in the next round
+func (p *Plugin) ReportRangesOutcome(
```

Comment:
nit: this is a pure function
```suggestion
func ReportRangesOutcome(
```